utils/utils.go

```diff
@@ -476,6 +482,24 @@ func GetSSHConfig() *ssh.ServerConfig {
 				}
 			}
 
+			// Allow validation of public keys via a sub-request to another service
+			authUrl := viper.GetString("authentication-key-request-url")
+			if authUrl != "" {
+				validKey, err := checkAuthenticationKeyRequest(authUrl, authKey)
```

Comment:
Thoughts on also sending things like client address or username? I feel like that would be easy to do and could widen the functionality here.

Reply:
Yep this is a great idea!

Reply:
I've added this and switched the request to use a JSON body that includes the public key, username, and remote address. Let me know what you think